### ISSUE TITLE
No login form after session expire

### ISSUE THREAD

**Issue by matjazpotocnik** [thetuningspoon/AdminPreSaveValidation#2]
When a module is active, and a session expires ($config->sessionExpireSeconds), and you're editing a page, and then you save the page, the login form is not displayed.

Here is a simple test that you can perform:

- Open two browser tabs. In one of them edit a page, in the other tab, go anywhere you want in the PW admin.
- In your second tab, log out.
- Return to your first tab and click the "Save" button. This should take you to the login screen.

Perhaps just adding canSubmit = true to the ajax fail event would be enough?

**Comment by thetuningspoon** [comment]
Thanks for the issue report. I’m not sure that this is a bad thing, since the alternative of redirecting to the login screen would mean that your changes would be lost. I’d rather have the option to log back in in another tab and then resubmit the page. Maybe it should pop up a js alert to say “It looks like you’ve been logged out”?

**Comment by matjazpotocnik** [comment]
Yes, a popup should inform the user that he is no longer logged in, along with instructions on what to do. The problem now is that hitting the Save button doesn't do anything, and the user doesn't know what to do. Ryan's new module (https://processwire.com/blog/posts/page-edit-restore-module/) is created just for those cases. Unfortunately, it can't do its job if your module is active. Could you check if PageEditRestore is installed and then just set canSubmit = true otherwise pop a warning as you suggested?
